src/main/java/mate/academy/springbookapp/service/order/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,94 @@
+package mate.academy.springbookapp.service.order.impl;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import mate.academy.springbookapp.dto.order.OrderDto;
+import mate.academy.springbookapp.dto.order.PlaceOrderRequestDto;
+import mate.academy.springbookapp.dto.order.UpdateStatusRequestDto;
+import mate.academy.springbookapp.dto.orderitem.OrderItemDto;
+import mate.academy.springbookapp.exception.EntityNotFoundException;
+import mate.academy.springbookapp.mapper.OrderItemMapper;
+import mate.academy.springbookapp.mapper.OrderMapper;
+import mate.academy.springbookapp.model.Order;
+import mate.academy.springbookapp.model.OrderItem;
+import mate.academy.springbookapp.model.ShoppingCart;
+import mate.academy.springbookapp.model.User;
+import mate.academy.springbookapp.repository.cartitem.CartItemRepository;
+import mate.academy.springbookapp.repository.order.OrderRepository;
+import mate.academy.springbookapp.repository.shoppingcart.ShoppingCartRepository;
+import mate.academy.springbookapp.service.order.OrderService;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final OrderItemMapper orderItemMapper;
+    private final OrderMapper orderMapper;
+
+    @Override
+    public List<OrderDto> getAllOrders(Long userId) {
+        return orderRepository.findAllByUserId(userId).stream()
+                .map(orderMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public List<OrderItemDto> getAllItemsFromOrder(Long userId, Long orderId) {
+        Order modelOrder = orderRepository.findByIdWithItems(orderId).orElseThrow(
+                () -> new EntityNotFoundException("Can't find order with id: " + orderId));
+        if (!modelOrder.getUser().getId().equals(userId)) {
+            return null;
+        }
```

Comment:
1. Avoid using `null` when possible.
2. Think of how should your application behave when user tries to access someone else's resources. I think that you should either return 404 (more secure - you're not telling anyone that resource exists, but may be misleading for debugging) or 403 (more explicit).